design/secure-communication.md

```diff
@@ -0,0 +1,94 @@
+# Secure Communication for Contour and Envoy
+
+Status: Draft
+
+This document outlines a design for securing gRPC communication between Contour and Envoy, including all the xDS endpoints.
+
+## Goals
+
+- Achieve secure communication between Contour and Envoy, which will allow a split deployment.
+- Allow use of the SDS service in Contour with authentication.
```

Comment:
SDS isn't special here, we're encrypting all the gRPC endpoints, not just SDS. 

Previous attempts at this design tried to use SDS over a loopback or unix socket to try to sidestep certs on disk, but there is nothing special about SDS in this application.

I'd drop this line.